includes/DataObjects/User.php

```diff
@@ -652,7 +654,11 @@ public function isAdmin()
     
 	public function isCheckuser()
 	{
-		return $this->checkuser == 1 || $this->oauthCanCheckUser();
+	    if($this->isCheckuserCache === null) {
```

Comment:
Miiiiinor style note: Space after the `if`.  Not really important though.  Fix it if you want, otherwise go ahead and merge.  Also mix of spaces and tabs, but *shrug*.